pkg/policy/report.go

```diff
@@ -49,64 +49,76 @@ func (pc *PolicyController) forceReconciliation(reconcileCh <-chan bool, cleanup
 	for {
 		select {
 		case <-ticker.C:
-			logger.Info("performing the background scan", "scan interval", pc.reconcilePeriod.String())
-			if err := pc.policyReportEraser.CleanupReportChangeRequests(cleanupReportChangeRequests, nil); err != nil {
-				logger.Error(err, "failed to cleanup report change requests")
-			}
+			if !toggle.DisableBackgroundScan.Enabled() {
+				logger.Info("performing the background scan", "scan interval", pc.reconcilePeriod.String())
+				if err := pc.policyReportEraser.CleanupReportChangeRequests(cleanupReportChangeRequests, nil); err != nil {
+					logger.Error(err, "failed to cleanup report change requests")
+				}
 
-			if err := pc.policyReportEraser.EraseResultEntries(eraseResultEntries, nil); err != nil {
-				logger.Error(err, "continue reconciling policy reports")
-			}
+				if err := pc.policyReportEraser.EraseResultEntries(eraseResultEntries, nil); err != nil {
+					logger.Error(err, "continue reconciling policy reports")
+				}
 
-			pc.requeuePolicies()
-			pc.prGenerator.MapperInvalidate()
+				pc.requeuePolicies()
+				pc.prGenerator.MapperInvalidate()
+			} else {
+				logger.Info("background scan is disabled")
+			}
 
 		case erase := <-reconcileCh:
-			logger.Info("received the reconcile signal, reconciling policy report")
-			if err := pc.policyReportEraser.CleanupReportChangeRequests(cleanupReportChangeRequests, nil); err != nil {
-				logger.Error(err, "failed to cleanup report change requests")
-			}
+			if !toggle.DisableBackgroundScan.Enabled() {
```

Comment:
Q: if background scan is disabled, shouldn't we allow the cleanup at least in case of force reconciliation ?

Reply:
I will trust you on this one, I will change it.

Reply:
@prateekpandey14 made changes. does it look better ?

Reply:
lgtm